### ISSUE TITLE
Deep Training with Tensorflow

### ISSUE THREAD

**Issue by javipa1995** [tensorflow/tensorflow#37044]
Hello,

i'm trying to make an experiment about performance with Keras and Tensorflow. I'm interested in training some parts of the networks. In my case, my objective is training some connections (neurons) of certains layers. Is it possible to do that experiment? I know that re-training layers is possible with trainable=True, but I would like to do it in a deeper way, going to the connections (fully connected phase) and filters (convolutional phase)

Thanks,

Javier

**Comment by jvishnuvardhan** [comment]
@javipa1995 Can you please describe your use case or the reason behind this specific experiment? Thanks!

**Comment by tanzhenyu** [comment]
We probably needs more context than this, for example, do you have existing example (written in TF), or the problems to solve?

**Comment by jvishnuvardhan** [comment]
@javipa1995 I think you can monitor filters and see how they get updated during training. There are couple of resources like [this](https://stackoverflow.com/questions/39361943/how-can-visualize-tensorflow-convolution-filters) and [this](https://machinelearningmastery.com/how-to-visualize-filters-and-feature-maps-in-convolutional-neural-networks/). You can find many more when you search.

TF website has lot of resources for performance. You can check [this resource](https://www.tensorflow.org/guide/profiler) to profile the model performance. In future, post this kind of questions in Stackoverflow as there is a huge community helping each other.

GitHub is mainly for bug and performance related issues. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing this as this was resolved with the above comment. Please feel free to reopen if I am mistaken. Thanks!

**Comment by pandrey-fr** [comment]
Hi,
While we lack context (and responses), I do not think that this is actually solved. If I understand well, the question is to know whether there is a mechanism to selectively set some of the weights within a layer trainable, while the other weights would not change. In other words, the question is to know whether there is a straightforward way to have a more granular approach of the `layer.trainable` API.
As far as I know, there is no such mechanism, but it could be achieved by selectively applying no-gradient ops, which would probably be somewhat tedious... @jvishnuvardhan - or anyone else on the TF team - would you have some additional thoughts, insights or knowledge as to that?
Thank you.
